### DISCUSSION TITLE
Global :pinAllExceptPeerDependencies, project specific :pinOnlyDevDependencies

### DISCUSSION THREAD

**Discussion by firecow** [renovatebot/renovate#18710] [Request Help] [answered]
We have `:pinAllExceptPeerDependencies` set in the global renovate config.
For our PHP library packages located in the `packagist` gitlab group, we only want to `:pinDevDependencies`

`# Output from renovate dry-run`
```
DEBUG: migrated config (repository=packagist/player-account-client)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "assignees": ["ok", "mma"],
         "extends": [
           "group:monorepos",
           "group:recommended",
           "workarounds:all",
           ":renovatePrefix",
           ":separateMultipleMajorReleases",
           ":pinAllExceptPeerDependencies",
           ":rebaseStalePrs",
           ":updateNotScheduled",
           ":pinOnlyDevDependencies"
         ]
       }
```

`# renovate.json located in packagist/player-account-client`
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [":pinOnlyDevDependencies"]
}
```

This config constellation doesn't bring up any MR's in `packagist/player-account-client` regarding pinning dev dependencies.

How would you guys achieve this this?

If it possible to use a shared config only on the `packagist/*` group, that would be very awesome. 
That way we don't have to add a project specific settings to each and every project in that group.

**Comment by viceice** [comment]
you can use ignore preset option in repo config to exclude specific presets.

**Reply by firecow** [reply]
I don't think that `composer.json` `require-dev` field is interpreted as a devDependency by Renovatebot.
I shall investigate.

**Comment by firecow** [comment, marked as answer]
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json", 
  "ignorePresets": [":pinAllExceptPeerDependencies"],
  "extends": [":pinOnlyDevDependencies"]
}
```
Closes all MR's requests about `composer.json` pinning

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "ignorePresets": [":pinAllExceptPeerDependencies"],
  "packageRules": [{ "matchDepTypes": ["require-dev"], "rangeStrategy": "pin" }]
}
```
This opens a MR pinning only `require-dev` types in `composer.json`, exactly how we want it.

Have I stumbled upon a bug, or is `require-dev` not supposed to be catched by `:pinOnlyDevDependencies` ?

**Reply by viceice** [reply]
not supported, here is the preset: https://docs.renovatebot.com/presets-default/#pinonlydevdependencies

you can probably open a pr to update this
https://github.com/renovatebot/renovate/blob/638302e519a3c76c01f185b22a4d05289f054b9f/lib/config/presets/internal/default.ts#L112-L113
and this
https://github.com/renovatebot/renovate/blob/638302e519a3c76c01f185b22a4d05289f054b9f/lib/config/presets/internal/default.ts#L98-L99
